### ISSUE TITLE
com.ctc.wstx.evt.CompactStartElement#getAttributes() inverts "specified" flag on attributes

### ISSUE THREAD

**Issue by kjherron** [FasterXML/woodstox#30]
CompactStartElement.getAttributes() returns a list of attributes where the isSpecified() flag is false for attributes which were specified in the source XML. This is backwards.

The attached sample [WS.java.txt](https://github.com/FasterXML/woodstox/files/638034/WS.java.txt) produces this output:

> com.ctc.wstx.stax.WstxInputFactory
> com.ctc.wstx.evt.WstxEventReader
> com.ctc.wstx.evt.CompactStartElement org.codehaus.stax2.ri.evt.AttributeEventImpl c d false

The "false" in the last line is the isSpecified() return value for the sample XML, and it ought to be true.

There are actually a couple of problems with handling of the "specified" flag for attributes. The relevant AttributeEventImpl ctor looks like this, implying the last argument should be true for specified attributes:

```
    public AttributeEventImpl(Location loc, String localName, String uri, String prefix,
                              String value, boolean wasSpecified)
```

However, CompactStartElement#constructAttr() has this:

```
    public Attribute constructAttr(String[] raw, int rawIndex, boolean isDef)
    {
        return new AttributeEventImpl(mLocation, raw[rawIndex], raw[rawIndex+1],
                                      raw[rawIndex+2], raw[rawIndex+3], isDef);
    }
```
The argument name "isDef" suggests the flag should be true if the flag is default, but it's passing it unchanged to the ctor which expects a value that's true if the flag is _not_ default. Then, in #getAttributes() we have this:
`                l.add(constructAttr(rawAttrs, i, (i >= defOffset)));
`
so it's passing a flag which is true if the attribute is default. However, in #getAttributeByName() we have this:
`        return constructAttr(mRawAttrs, ix, !mAttrs.isDefault(ix));
`
So it's passing a flag which is false if the attribute is default.

We originally found this with Woodstox 4.2.1. I can also reproduce it with 5.0.3.

**Comment by cowtowncoder** [comment]
That does seem wrong, and I do not see a reason why this should not be inverted.
And name `isDef` is particularly poorly chosen since it could mean either "is default" OR "is (explicitly) defined".

As to this: 

```java
l.add(constructAttr(rawAttrs, i, (i >= defOffset)));
```

I think that actually looks correct since (if I remember correctly) values for default attributes are prepended first... 

Anyway; what would be very useful would be a simple unit test.

**Comment by cowtowncoder** [comment]
Actually I think this was fixed as #43.
